### PR TITLE
User Description Screen UI (badge for counts)

### PR DESCRIPTION
**Pull request by kostokord**

### CODE REVIEW

File: Degrees/AppDelegate.swift [hunk 1]
```diff
@@ -16,9 +16,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         let loginViewController = LoginViewController()
+        let userProfileViewController = UserProfileViewController()
         
         rootNavigationController = UINavigationController()
-        rootNavigationController?.setViewControllers([loginViewController], animated: false)
+        rootNavigationController?.setViewControllers([userProfileViewController], animated: false)
```
Comment (oliromole) [inline Degrees/AppDelegate.swift]: Remove changes, because it is temporary code.

File: Degrees/Controllers/UserProfile/UserProfileView.swift [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        func showLabels (_ userStatus: UserStatus){
+                countActiveLabel.text = "\(userStatus.activeQuizCnt)"
+                drawLabel(countActiveLabel)
+                countClosedLabel.text = "\(userStatus.completeQuizCnt)"
+                drawLabel(countClosedLabel)
+                timer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(timeToRefreshLabelUpdate), userInfo: nil, repeats: true)
+        }
+        
+        private func drawLabel(_ label: UILabel) {
+                label.textColor = .red
```
Comment (omurashchenko-onseo) [inline Degrees/Controllers/UserProfile/UserProfileView.swift]: Use config (e.g. filaprivate struct ViewConfig with all the settings) for all appearance settings

File: Degrees/Controllers/UserProfile/UserProfileView.swift [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        
+        @IBAction func quizListButtonPressed(_ sender: UIButton) {
+                timer.invalidate()
+                listenerUserProfile?.onQuizListButtonPressed(button: sender)
+        }
+        
+        @objc func timeToRefreshLabelUpdate(){
+                timeToRefresh -= 1
+                if timeToRefresh == 0 {
+                        listenerUserProfile?.pullToRefresh()
```
Comment (omurashchenko-onseo) [inline Degrees/Controllers/UserProfile/UserProfileView.swift]: it's not pull to refresh, name the method correctly

File: Degrees/Controllers/UserProfile/UserProfileView.swift [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                        subordinatesIdLabel.text = managerId
+                }
+                
+                if #available(iOS 10.0, *){
+                        refreshScrollView.refreshControl = refreshData
+                } else {
+                        refreshScrollView.addSubview(refreshData)
+                }
+                
+                refreshData.tintColor = #colorLiteral(red: 0.0007662897115, green: 0.80061692, blue: 0.9980433583, alpha: 1)
```
Comment (omurashchenko-onseo) [inline Degrees/Controllers/UserProfile/UserProfileView.swift]: use config with constants

File: Degrees/Controllers/UserProfile/UserProfileView.swift [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        }
+        
+        @IBAction func quizListButtonPressed(_ sender: UIButton) {
+                timer.invalidate()
+                listenerUserProfile?.onQuizListButtonPressed(button: sender)
+        }
+        
+        @objc func timeToRefreshLabelUpdate(){
+                timeToRefresh -= 1
+                if timeToRefresh == 0 {
```
Comment (omurashchenko-onseo) [inline Degrees/Controllers/UserProfile/UserProfileView.swift]: it's safer to use <= 0

File: Degrees/Controllers/UserProfile/UserProfileViewController.swift [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+//
+
+import Foundation
+import UIKit
+
+class UserProfileViewController: UIViewController, ListenerUserProfile {
+        
+        var userProfileView: UserProfileView!
+        var userProfile = UserProfile(id: "", firstName: "", lastName: "", avatarUrl: "", managerId: [], subordinatesId: [])
+        var userStatus = UserStatus(activeQuizCnt: 5, completeQuizCnt: 4)
```
Comment (omurashchenko-onseo) [inline Degrees/Controllers/UserProfile/UserProfileViewController.swift]: do not commit dummy data for common branche

File: Degrees/Controllers/UserProfile/UserProfileViewController.swift [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        override func viewDidLoad() {
+                super.viewDidLoad()
+                loadDataFromJSON()
+                refreshData()
+                userProfileView.updateView(userProfile)
+                userProfileView.showLabels(userStatus)
+        }
+        
+        func refreshData(){
+                let auth = AuthUserRequest(login: "51", password: "pass51")
```
Comment (omurashchenko-onseo) [inline Degrees/Controllers/UserProfile/UserProfileViewController.swift]: do not commit dummy data for common branch

File: Degrees/Controllers/UserProfile/UserProfileViewController.swift [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+                                self.userProfileView.isWorkIndikator(isAnimated: false, indicator: self.userProfileView.actIndicator)
+                        }
+                }
+        }
+        
+        func loadDataFromJSON() {
+                if let path = Bundle.main.path(forResource: "userProfile", ofType: "json") {
+                        do {
+                                let data = try Data(contentsOf: URL(fileURLWithPath: path), options: .mappedIfSafe)
+                                let jsonResult = try JSONSerialization.jsonObject(with: data, options: .mutableLeaves)
```
Comment (omurashchenko-onseo) [inline Degrees/Controllers/UserProfile/UserProfileViewController.swift]: Use JSONDecoder instead of JSONSerialization

**Pull request merged**